### PR TITLE
Add new tests for `dpnp.einsum` and adopt it with `NumPy-2.0.0`

### PR DESCRIPTION
**Pull request by vtavana**
In this PR, 

1) new tests are added for `dpnp.einsum` and accordingly a few updates are made in the code to pass all the test.
2) The tests are update to pass with any numpy version between 1.23 and 2.0.
3) docsting of several functions are updated to be consistent with each other.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_utils/dpnp_utils_einsum.py [hunk 1]
```diff
@@ -853,6 +800,12 @@ def _parse_ellipsis_subscript(subscript, idx, ndim=None, ellipsis_len=None):
 def _parse_int_subscript(list_subscript):
     """Copied from _parse_int_subscript in cupy/core/_einsum.py"""
 
+    if not isinstance(
+        list_subscript, (list, tuple, numpy.ndarray, dpnp.ndarray)
```
Comment (npolina4) [inline dpnp/dpnp_utils/dpnp_utils_einsum.py:804]: Do we want to work with individual array here or do we want to work with the list?
Please change the input argument check or error description.

File: dpnp/dpnp_utils/dpnp_utils_einsum.py [hunk 2]
```diff
@@ -865,6 +818,11 @@ def _parse_int_subscript(list_subscript):
                     "For this input type lists must contain "
                     "either int or Ellipsis"
                 ) from e
+            if isinstance(s, int):
+                if not 0 <= s < len(_einsum_symbols):
+                    raise ValueError(
+                        "subscript is not within the valid range [0, 52)."
```
Comment (npolina4) [inline dpnp/dpnp_utils/dpnp_utils_einsum.py:824]:
```suggestion
                        f"subscript is not within the valid range [0, {len(_einsum_symbols)})."
```

File: dpnp/dpnp_utils/dpnp_utils_einsum.py [hunk 3]
```diff
@@ -1012,8 +970,14 @@ def _transpose_ex(a, axeses):
         strides.append(stride)
 
     # TODO: replace with a.view() when it is implemented in dpnp
-    a = _view_work_around(a, shape, strides)
-    return a
+    return dpnp_array(
```
Comment (npolina4) [inline dpnp/dpnp_utils/dpnp_utils_einsum.py:973]: Сan we delete this comment?

Comment (vtavana) [reply dpnp/dpnp_utils/dpnp_utils_einsum.py:973]: When `dpnp.view` is implemented we can directly use it here so I guess it is better to keep the comment for now.

**Review by npolina4 — APPROVED** [review]
Thank you @vtavana!

**Pull request merged**